docs/TIGER_STYLE.md

```diff
@@ -38,6 +38,14 @@ An hour or day of design is worth weeks or months in production:
 
 > “the simple and elegant systems tend to be easier and faster to design and get right, more efficient in execution, and much more reliable” — Edsger Dijkstra
 
+### Tools
+
+Often, the way to simplicity is through tooling. Computers are great at doing repetitive, routine tasks. We use our imagination to take advantage of that. Writing perfect code is hard. Writing a simulator which automatically and relentlessly seeks for inputs which uncover code's imperfections is much simpler.
+
+Tools have costs. A single universal toolbox is simpler to operate than an array of ultra-specialized instruments each coming with a dedicated manual. Our primary tool is Zig -- it's not the best for everything, but is good enough for most of the things. We invest into our Zig toolbox to ensure we can tackle new problems quickly.
+
+> “The right tool for the job is often the tool you are already using -- adding new tools has a higher cost than many people appreciate” — John Carmack
+
```

Comment:
The sections on simplicity, elegance, technical debt are tied together by the introduction, `First, a word on simplicity, elegance and technical debt.`

Therefore, let's rather group this section on `Tooling` together with a new section on `Dependencies`. The zero-dependencies bullet is currently under `Cache Invalidation`, but could really be its own section, and I think that would then lead well into `Tooling`.

We can put both of these just after `Style by the numbers`, before `The Last Stage`:

```
### Dependencies

TigerBeetle has a “zero dependencies” policy, apart from the Zig toolchain. Dependencies, in general, inevitably lead to supply chain attacks, safety and performance risk, and slow install times. For foundational infrastructure in particular, the cost of any dependency is further amplified throughout the rest of the stack.

### Tooling

Similarly, tools have costs. A small standardized toolbox is simpler to operate than an array of specialized instruments each with a dedicated manual. Our primary tool is Zig. It may not be the best for everything, but it's good enough for most things. We invest into our Zig tooling to ensure that we can tackle new problems quickly, with a minimum of local development enviroment complexity.

> “The right tool for the job is often the tool you are already using—adding new tools has a higher cost than many people appreciate” — John Carmack

For example, the next time you write a script, instead of `scripts/*.sh`, write `scripts/*.zig`.

This not only makes your script cross-platform and portable, but introduces type safety and increases the probability that running your script will succeed for everyone on the team, instead of hitting a Bash/Shell/OS-specific issue.

Standardizing on Zig for tooling is important to ensure that we reduce tooling dimensionality as the team, and therefore the range of personal tastes, grows. This may be slower for you in the short term, but makes for more velocity for the team in the long term.
```

Reply:
Yeah, I've struggled a bit with finding an appropriate place for this. Zero dependency indeed fits better with "simplicity" than with "cache invalidation". 

In general, our tiger style is very poetic, which is very inspirational, but at the same time makes it hard to amend it with specific rules we discover over time. I wounder if we  perhaps should split the doc here into two parts:

* poetic principles with quotes and music
* list of rules

List of rules would be an instance of [kitchen sink](https://matklad.github.io/2021/05/12/design-pattern-dumping-ground.html) meta pattern for creating an obvious place to add poorly categorizable things to.

Though, deps&tooling definitely belong to principles section.

Reply:
Let's leave it as is for now, thanks @matklad—the structure of the doc is safety/performance/experience, which is the big pattern for how it's all grouped. If we need to tweak structure later we can!

Reply:
Just need to remove the duplicate "Tools" section up top, and then this is good to go!

Reply:
:facepalm: removed